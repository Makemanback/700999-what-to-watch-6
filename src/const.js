export const FILMS_ON_SCREEN = 8;
export const ratings = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const Genre = {
  ALL_GENRES: `All genres`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMAS: `Dramas`,
  HORROR: `Horror`,
  KIDS_FAMILY: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`
};


export const Path = {
  DEFAULT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  FILM_ID: `/films/:id`,
  FILM_REVIEW: `/films/:id/review`,
  MOVIE_REVIEWS: `/films/:id/reviews`,
  PLAYER: `/player/:id`,
  MOVIE_DETAILS: `/films/:id/details`
};

export const Level = {
  AWESOME: `Awesome`,
  VERY_GOOD: `Very Good`,
  GOOD: `Good`,
  NORMAL: `Normal`,
  BAD: `Bad`,
};
