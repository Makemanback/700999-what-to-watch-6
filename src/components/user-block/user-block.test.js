import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';

import {AuthorizationStatus} from '../../const';

import UserBlock from './user-block';

const mockStore = configureStore({});
test(`Should UserBlock render correctly`, () => {

  const store = mockStore({
    USER: {authorizationStatus: AuthorizationStatus.AUTH}
  })

  const history = createMemoryHistory();
  const {container} = render(
    <redux.Provider store={store}>
      <Router history={history}>
        <UserBlock />
      </Router>
    </redux.Provider>
  );

  expect(container).toMatchSnapshot();
});
