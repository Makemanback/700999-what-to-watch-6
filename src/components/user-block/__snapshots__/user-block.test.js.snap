// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should UserBlock render correctly 1`] = `
<div>
  <div
    class="user-block"
  >
    <a
      class="user-block__link"
      href="/mylist"
    >
      <div
        class="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="img/avatar.jpg"
          width="63"
        />
      </div>
    </a>
  </div>
</div>
`;
